survey-sdk/src/main/java/co/nimblehq/survey/sdk/SurveyApi.kt

```diff
@@ -0,0 +1,52 @@
+package co.nimblehq.survey.sdk
+
+import co.nimblehq.survey.sdk.api.AppService
+import co.nimblehq.survey.sdk.entity.SurveyEntity
+import co.nimblehq.survey.sdk.network.NetworkBuilder
+import moe.banana.jsonapi2.ArrayDocument
+import moe.banana.jsonapi2.ObjectDocument
+
+class SurveyApi private constructor(): NetworkBuilder() {
+    private val service: AppService by lazy { buildService() }
+    private var version = "v1"
+
+    companion object {
+        private val instance: SurveyApi by lazy { SurveyApi() }
+    }
+
+    fun withMode(debugMode: Boolean): SurveyApi {
+        setDebugMode(debugMode)
+        return this
+    }
+
+    fun withBaseUrl(url: String): SurveyApi {
+        setBaseUrl(url)
+        return this
+    }
+
+    fun withConnectionTimeout(timeout: Long): SurveyApi {
+        setConnectionTimeoutInSecond(timeout)
+        return this
+    }
+
+    fun withReadTimeout(timeout: Long): SurveyApi {
+        setReadTimeoutInSecond(timeout)
+        return this
+    }
+
+    fun withVersion(version: String): SurveyApi {
+        this.version = version
+        return this
+    }
+
+    //below are public apis
+
+    suspend fun getSurveyList(page: Int, size: Int): ArrayDocument<SurveyEntity> {
```

Comment:
So developers have to handle background jobs with coroutine? 🤔  What happens if the application is applying Rx instead? I'd suggest getting rid of handling background jobs inside the SDK, the application would handle by itself.

cc @luongvo

Reply:
@hoangnguyen92dn Do you have anh idea to make it general?
I was suggested using Coroutine template and I think for the first version, the SDK will support with Coroutine only.
We will think about a better way in next phase. Are you ok with this @hoangnguyen92dn @luongvo ?

Reply:
@doannimble How about getting rid of `suppend` from `SurveyApi` and `AppService`? The sample app needs to call the api with CoroutineScope 🤷‍♂️

Reply:
@hoangnguyen92dn It's a big problem, as you know with RxJava, we have to wrap up the result with `Observable/Flowable`. 
I think the app need to control its lifecycle and thread. We should not make more complex things on this phase 🙏

Reply:
@doannimble 

> we have to wrap up the result with Observable/Flowable.

🤷‍♂️  I don't think so. By using this SDK, the application is forced to use Coroutine? I mean that the SDK should return the raw data rather than `suspend` or `Obsservable` here.

Reply:
@hoangnguyen92dn I think we should ask @sleepylee about the scope of this SDK.

Reply:
perhaps then we can send the data by using a callback? `onDataAvailable` or something like that so we could process the data in the background with our choice of networking solution?
Have you guys done any research for a sample reference?

Also for the comment, I'd recommend to add the comments like you often see:

```
/*
  * Return the list of Surveys. The request is asynchronously done in the background.
  *
  * params: [if there is]
  * returns: [List<Survey>] responded from the backend.
getSurveys(): List<Survey>;
```

Reply:
@sleepylee Okay, let me update the way to return data